### ISSUE TITLE
Feature request: decode very common encodings

### ISSUE THREAD

**Issue by shirish93** [sqlitebrowser/sqlitebrowser#501]
Hello Sqlitebrowser team!

I've been using Sqlite + SqliteBrowser to replace a very expensive only MongoDB service. It's been working great for me, but most of my work consists of 'dumping' data and retrieving it, without a lot of complex queries. Thus, I've been using @piskvorksky's SqliteDict (for python), which 'pickles' the values and stores them in the table. 

While I am not particularly connected to 'pickling', it would be really great if there were a 'decoding' of most common encodings in the fields. For example, one might dump entire ProtoBuf (google's)  objects into the database, and be able to see 'real-time' changes, as it is decoded by the viewer.

In a verry related note, would there be a possibility of creating an extension environment from other languages? I could write the extension I'm suggesting in Python, but would have a hard time in c++.

Thank you!

**Comment by justinclift** [comment]
Yeah, an extension environment would be great.  I don't know if we have anyone around who could implement it though.  If someone wants to volunteer/join-the-team/etc, that would be fantastic. :grinning:

**Comment by shirish93** [comment]
Now that you mention, I might try to get my hands on trying to implement my request. The things that I learn, I'll document, it'll be useful as a foundation for other extension-makers.

I just need onee hint to get started. I'm trying to figure out the point between the query returning results, and the results being displayed, but have been unable to do so. Any idea what file/lines I should be looking at?

Somewhat related, perhaps it might be a good idea for someone more knowledgeable to add a little bit of documentation, if only in the header files, and at a file level? _That_ would definitely be a great first step towards starting a healthy extension environment.

**Comment by justinclift** [comment]
@rp- Are you able to point out the code between the query result and the display, as @shirish93 is asking?  If so, that would help him get start in the code base... :smile:

@shirish93 The other approach would be to just set a bunch of breakpoints in the source, then step through them as the query happens.  It's not the fastest way to narrow things down, but it would get you to the code you're looking for within a few minutes.  And after that, you know where the code is, so don't need to do it again. :smile:

**Comment by MKleusberg** [comment]
There are two places (leaving out a few unimportant ones) for issuing queries and displaying them in the UI: the Browse Data tab and the Execute SQL tab. For the latter I don't see any real chance of implementing some automatic decoding procedure. But for the former one we already have some sort of mechanism which might not be enough, though. You can have a look at the 'display formats' part of the code, especially in the MainWindow.cpp and ColumnDisplayFormatDialog.cpp. My first opinion would be that writing a SQLite extension and using that in the display format dialog would make most sense. But either way, we'd have to check for some C implementation of (un)pickle.
